zerver/lib/actions.py

```diff
@@ -3508,3 +3509,29 @@ def do_update_user_custom_profile_data(user_profile, data):
             update_or_create(user_profile=user_profile,
                              field_id=field['id'],
                              defaults={'value': field['value']})
+
+def get_unread_message_ids_per_recipient(user_profile):
+    # type: (UserProfile) -> List[Tuple[Union[Text, List[Dict[str, Any]]], List[int]]]
+    user_msgs = UserMessage.objects.filter(
+        user_profile=user_profile,
+        flags=~UserMessage.flags.read,
+    ).order_by(
+        'message__recipient__id',
+    ).values(
+        'message__id',
+        'message__recipient__id',
+        'message__recipient__type',
+        'message__recipient__type_id',
```

Comment:
I will update this to include the topic. While I am at it I will include the `recipent_id`s in the result for the web. The format will look like this:
```
[   
    [   
        {   
            'display_recipient': 'general',
            'recipient_id': 54,
            'subject': 'lunch',
        },
        [400, 823]
    ],
    ...
]
```